src/main/java/noel/helper/Parser.java

```diff
@@ -0,0 +1,116 @@
+package noel.helper;
+
+import noel.tasks.Task;
+
+public class Parser {
+
+    private final Tasklist tasks;
+    private final Storage storage;
+
+    public Parser(Tasklist tasks, Storage storage) {
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    public static void printFunction(String message){
+        String filler = "____________________________________________________________";
+        System.out.println(filler + "\n" + message + "\n" + filler);
+    }
+
+    public int parseCommand(String nextLine) {
+        String command;
+        if (nextLine.equals("bye")) {
+            return -1;
+        } else if (nextLine.equals("list")) {
+            tasks.printTaskList();
+        } else if (nextLine.contains(" ")) {
+
+            String[] result = nextLine.split(" ");
+            command = result[0];
+
+            switch (command) {
+
+                case "mark": {
+                    int taskNum = Integer.parseInt(result[1]);
+                    taskNum = taskNum - 1;
+                    tasks.markAsDone(taskNum);
+                    storage.writeToFile(tasks.getTaskAsList());
+                    break;
+                }
+                case "unmark": {
+                    int taskNum = Integer.parseInt(result[1]);
+                    tasks.unMark(taskNum);
+                    storage.writeToFile(tasks.getTaskAsList());
+                    break;
+                }
+                case "todo":
```

Comment:
there is some inconsistency in using of the curly braces `{`, some case statements include them and some does not. Maybe some consistency will be good